fedimint-client/src/sm/executor.rs

```diff
@@ -390,181 +410,235 @@ impl ExecutorInner {
-                    .await
-                    .map_err(|e| match e {
-                        AutocommitError::CommitFailed {
-                            last_error,
-                            attempts,
-                        } => last_error
-                            .context(format!("Failed to commit after {attempts} attempts")),
-                        AutocommitError::ClosureError { error, .. } => error,
-                    })?;
-
-                // TODO: add a INFO version without all the details
-                debug!(
-                    outcome = ?active_or_inactive_state,
-                    "Finished executing state transition"
-                );
-
-                active_state_count -= 1;
-                match active_or_inactive_state {
-                    ActiveOrInactiveState::Active {
-                        dyn_state,
-                        active_state,
-                    } => {
-                        if let Some(transition) = self
-                            .get_transition_for(dyn_state.clone(), active_state, global_context_gen)
-                            .await
-                        {
-                            active_state_count += 1;
-                            transitions.push(transition);
+                                ActiveOrInactiveState::Inactive { dyn_state } => {
+                                    notifier.notify(dyn_state.clone());
+                                }
+                            }
+                            ExecutorLoopEvent::Completed { state, outcome }
                         }
-                        self.notifier.notify(dyn_state);
-                    }
-                    ActiveOrInactiveState::Inactive { dyn_state } => {
-                        self.notifier.notify(dyn_state);
-                    }
+                        .instrument(span),
+                    )});
+                }
+                ExecutorLoopEvent::Completed { state, outcome } => {
+                    currently_running_sms.remove(&state);
+                    info!(
+                        target: LOG_CLIENT_REACTOR,
+                        operation_id = %state.operation_id(),
+                        outcome_active = outcome.is_active(),
+                        total = futures.len(),
+                        "State transition complete");
+                    trace!(
+                        target: LOG_CLIENT_REACTOR,
+                        ?outcome,
+                        operation_id = %state.operation_id(), total = futures.len(),
+                        "State transition complete");
+                }
+                ExecutorLoopEvent::Disconnected => {
+                    info!(target: LOG_CLIENT_REACTOR, "Disconnected. Terminating.");
+                    break;
                 }
-                anyhow::Ok(())
             }
-            .instrument(span)
-            .await?
         }
+
+        trace!(target: LOG_CLIENT_REACTOR, "Terminated");
```

Comment:
What's the reason for having both these logs at these two levels?

Reply:
Sloppiness.